### ISSUE TITLE
Multiprocessing: Unable to open file

### ISSUE THREAD

**Issue by samplestr** [Roche/pyreadstat#85]
Describe the issue
Try to use the new module read_file_multiprocessing for the sample.sav and im getting "ReadstatError: Unable to open file"
______________________________________________
Error: 
![image](https://user-images.githubusercontent.com/74183459/98532116-8b696d80-2281-11eb-9900-e95549450139.png)
______________________________________________
To Reproduce:
import pandas as pd
import pyreadstat
______________________________________________
df, meta = pyreadstat.read_file_multiprocessing(pyreadstat.read_sav,'C:/Users/wuttk/Downloads/sample.sav')
______________________________________________
(df, meta = pyreadstat.pyreadstat.read_sav('C:/Users/wuttk/Downloads/sample.sav') <-- is working fine, )
______________________________________________
File example
sample.sav from the test repo
______________________________________________
Setup Information:
How did you install pyreadstat? (directly from repo)
Platform (windows 64 bit)
Python Version (3.7)
Python Distribution (Anaconda)
Using condaenv?

**Comment by ofajardo** [comment]
hi 

thanks for the detailed report. I cannot reproduce it, for me on windows works fine. The error is really weird since read_sav works I don't understand why the multiprocessing does not

In order to diagnose what is happening, could you try:
1- Can you install pyreadstat with pip (or maybe with conda)? I installed with pip and works fine. What do you mean by installed directly from repo? Did you compile yourself? maybe something is wrong during the compilation?
2- If you do this, does it work?:
```
# first cd into C:/Users/wuttk/Downloads/
df, meta = pyreadstat.read_file_multiprocessing(pyreadstat.read_sav, 'sample.sav')
```
3- Where are you executing this? I mean what is your working directory? Maybe you are executing from a remote shared
drive or something like that?
4- Can you replace the forward slashes by back slashes and put an r in front? r'C:\Users\wuttk\Downloads\sample.sav'
You could also use double back slashes instead.

**Comment by samplestr** [comment]
Hello again, 

1.1  I used the conda prompt-> used pip install pyreadstat->Successfully....-1.0.3
1.2 Also tried pip without conda -> same result
2. Doesn't work
3. No, nothing like this; just basic directory on my laptop
4. Tried it before, did also not work.
--> Did try it again on an fresh PC same negativ result.

Cheers my friend! Thanks in advance, you already helped me so much!

**Comment by ofajardo** [comment]
Sooo strange ...

How do you run your code? Do you use any kind of IDE? Does it work if you use a vanilla python interpreter on command line (anaconda shell)? That's how I tested.

**Comment by samplestr** [comment]
- JNotebook, VsCode
- with anaconda shell i got a different error but same reason: 
![image](https://user-images.githubusercontent.com/74183459/98650349-0e013400-2339-11eb-864c-5b12c2d8d924.png)

I will try to re implement it maybe my-self and will give you an update about my learnings.

**Comment by ofajardo** [comment]
OK at least the error looks a bit less mysterious this time. Well, still mysterious to me actually, but at least I can find something about it in stackoverflow, check this, maybe it helps, it seems you need an if "__name__" == __main__: thing in your script:

https://stackoverflow.com/questions/18204782/runtimeerror-on-windows-trying-python-multiprocessing

**Comment by ofajardo** [comment]
Right, now I can reproduce the issue. It was working before because I was doing it in the interactive console. If I write a script like this:

```
import pyreadstat
df, meta = pyreadstat.read_file_multiprocessing(pyreadstat.read_sav, 'sample.sav')
```

I see the error, but if I do this, the error is gone:

```
import pyreadstat

if "__name__" == "__main__":
     df, meta = pyreadstat.read_file_multiprocessing(pyreadstat.read_sav, 'sample.sav')
```

IT works

**Comment by samplestr** [comment]
Can't confirm, I had the same idea but still doesn't work.
btw it should be:  'if __name__ == "__main__":' 
  instead of:  'if  "__name__" == __main__:'

Still works for pyreadstat.read_sav; so the problem should be in the neighborhood of the pool function / worker/

**Comment by ofajardo** [comment]
How are you launching the script? try launching it from the anaconda promt console, that was working for me. If it would work then you would know it is something with your IDE.

**Comment by samplestr** [comment]
Solved

**Comment by ofajardo** [comment]
Can you please share the solution in case someone else faces the same issue?

**Comment by samplestr** [comment]
Oh sorry, forgot to send it before closing:
I had to uninstall pyreadstat two times and install in fresh again. 
Now it is working as a script:

**import pyreadstat
if __name__ == "__main__":
    df, meta = pyreadstat.read_file_multiprocessing(pyreadstat.read_sav, 'sample.sav')**

Thank you for your help!

**Comment by ofajardo** [comment]
@samplestr I detected a problem that could be causing issues with this (#87) , now solved in version 1.0.5 if you want to give it a try.
